src/monitors/FuncMonitorUtil.v3

```diff
@@ -0,0 +1,48 @@
+private class FuncEnterNoLoopProbe(probe: Probe) extends Probe {
+	var frameAccessor: FrameAccessor;
+	def fire(func: WasmFunction, pc: int, frame: TargetFrame) -> Resumption {
+		var curAccessor = frame.getFrameAccessor();
+		if (curAccessor == frameAccessor) return Resumption.Continue;
+		frameAccessor = curAccessor;
+		return probe.fire(func, pc, frame);
+	}
+}
+private class FuncEnterLoopProbe(probe: Probe) extends Probe {
```

Comment:
I think you can avoid a search of the stack of accessors if you have the exit probe pop. So you have a matched pair of enter/exit helper probes for the loop case, and none for the no-loop case.

Reply:
Makes sense, I'll try out that approach instead